### PR TITLE
Add systemd-analyze condition

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/analyze/analyze-condition.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+#include "service.h"
+
+typedef struct condition_definition {
+        const char *name;
+        ConfigParserCallback parser;
+        ConditionType type;
+} condition_definition;
+
+static const condition_definition condition_definitions[] = {
+        {"ConditionPathExists",             config_parse_unit_condition_path,   CONDITION_PATH_EXISTS},
```
Comment (poettering) [inline src/analyze/analyze-condition.c]: usually we'd place a space after the { and before the }

File: src/analyze/analyze-condition.c [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        Condition **target;
+
+        if ((p = startswith(line, "Condition")))
+                target = &u->conditions;
+        else if ((p = startswith(line, "Assert")))
+                target = &u->asserts;
+        else
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot parse \"%s\".", line);
+
+        for (unsigned i = 0; i < ELEMENTSOF(condition_definitions); i++) {
```
Comment (poettering) [inline src/analyze/analyze-condition.c]: hmm, the ELEMENTSOF() macro returns a size_t entity, and hence I wouldn't cast this needlessly to `unsigned` here, and instead just size `size_t i` as iterator

File: src/analyze/analyze-condition.c [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                if (r < 0)
+                        return r;
+        }
+
+        r = condition_test_list(u->asserts, assert_type_to_string, log_helper, u);
+        if (u->asserts)
+                log_notice("Asserts %s.", r > 0 ? "succeeded" : "failed");
+
+
+        q = condition_test_list(u->conditions, condition_type_to_string, log_helper, u);
```
Comment (poettering) [inline src/analyze/analyze-condition.c:150]: spurious double newline

File: src/analyze/analyze-condition.c [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                return log_error_errno(r, "Failed to initialize manager: %m");
+
+        log_debug("Starting manager...");
+        r = manager_startup(m, NULL, NULL);
+        if (r < 0)
+                return r;
+
+        r = unit_new_for_name(m, sizeof(Service), "test.service", &u);
+        if (r < 0)
+                return r;
```
Comment (poettering) [inline src/analyze/analyze-condition.c]: iirc this needs logging

**Comment by keszybz** [comment]
Thank you for the review. Updated.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**